api/v1/serviceinstance_validating_webhook.go

```diff
@@ -34,19 +35,37 @@ func (si *ServiceInstance) SetupWebhookWithManager(mgr ctrl.Manager) error {
 		Complete()
 }
 
-// +kubebuilder:webhook:verbs=delete,path=/validate-services-cloud-sap-com-v1-serviceinstance,mutating=false,failurePolicy=fail,groups=services.cloud.sap.com,resources=serviceinstances,versions=v1,name=vserviceinstance.kb.io,sideEffects=None,admissionReviewVersions=v1beta1;v1
+// +kubebuilder:webhook:verbs=delete;update;create,path=/validate-services-cloud-sap-com-v1-serviceinstance,mutating=false,failurePolicy=fail,groups=services.cloud.sap.com,resources=serviceinstances,versions=v1,name=vserviceinstance.kb.io,sideEffects=None,admissionReviewVersions=v1beta1;v1
 
 var _ webhook.Validator = &ServiceInstance{}
 
+// log is for logging in this package.
+var serviceinstancelog = logf.Log.WithName("serviceinstance-resource")
+var allowMultipleTenants bool
+
+func SetAllowMultipleTenants(isAllowed bool) {
+	allowMultipleTenants = isAllowed
+}
+
 func (si *ServiceInstance) ValidateCreate() (warnings admission.Warnings, err error) {
+	serviceinstancelog.Info("validate create", "name", si.Name)
+	if !allowMultipleTenants && si.Spec.SubaccountID != "" {
+		return nil, fmt.Errorf("using multiple subaccounts is not allowed")
+	}
 	return nil, nil
 }
 
 func (si *ServiceInstance) ValidateUpdate(old runtime.Object) (warnings admission.Warnings, err error) {
+	serviceinstancelog.Info("validate update", "name", si.Name)
+	oldInstance := old.(*ServiceInstance)
+	if oldInstance.Spec.SubaccountID != si.Spec.SubaccountID {
+		return nil, fmt.Errorf("subaccountID can not be changed")
```

Comment:
return nil, fmt.Errorf("changing the subaccountID for an existing instance is not allowed")

Reply:
done